redesign: add initial implementation of design for docs and indexes

willpoint opened the pull request.

willpoint committed:
redesign: add initial implementation of design for docs and indexes

odeke-em commented on _content/js/site.js:
```diff
@@ -256,26 +298,35 @@ window.initFuncs = [];
+  async function getLatestSortedGoResources() {
+    try {
+      const versionData = await (await fetch('/dl/?mode=json')).json();
+      if (!versionData.length) return [];
```
Please protect against null dereferences by
`if (!(versionData && versionData.length)) return [];`

odeke-em commented on cmd/golangorg/server.go:
```diff
@@ -57,7 +56,7 @@ var (
 	contentDir = flag.String("content", "", "path to _content directory")
 
-	runningOnAppEngine = os.Getenv("PORT") != ""
+	runningOnAppEngine = false // os.Getenv("PORT") != ""
```
We can infer this more intentionally by creating a new flag
```go
localrun := flag.Bool("localrun", false, "if running the server locally")

runningOnAppEngine := os.Getenv("PORT") && !*localRun 
```

odeke-em commented on cmd/golangorg/server.go:
```diff
@@ -75,13 +74,13 @@ func main() {
-		}
-		*contentDir = filepath.Join(repoRoot, "_content")
-	}
+	// if *contentDir == "" && !runningOnAppEngine {
```
Could you please leave this code as is given that we shall detect if a localrun? Or rather how come you commented out this code?

willpoint replied:
The commenting out is expected to be temporary. This was when Jamal was helping debug why the container was not working. After this, and after changing the dockerfile listen address to 0.0.0.0:port did it work. But using localrun is a better choice in retrospect.

odeke-em reviewed (changes requested):
Nice work @willpoint! First round of review, please take a look.

willpoint committed:
complete the download journey

willpoint committed:
fix issue with notification bar showing when not needed

willpoint committed:
fixes relating to feedback
- ensure left index is properly aligned
- change text to 'Take a tour of Go' instead of the less positive call to action previously used
- consistently use MacOS instead of darwin
- point download link to go to doc/install page
- move download link to the right end of the top nav

willpoint committed:
improve left index
- reduce left margin for level 1 subtree
- reduce bottom margin for role=group items

willpoint committed:
mobile: add responsive features
- add a sidebar that can be toggled by a button
- hide TOC on mobile view

willpoint committed:
menu: improve mobile
- ensure code element can be broken into multiple lines
- ensure span color of depth 2 on outline is same color as a element
- ensure when outside the menu trigger is clicked the outline is hides
  so users don't always have to click the outline menu to hide it

willpoint committed:
handle docs page ux
- make doc from top banner clickable without a dropdown
- create a page for /doc to hold blocks of key topics
- make arm64 the default for macos as newer devices come
  with arm64

willpoint committed:
update editor and tools menu

willpoint committed:
docs: create content for editors and tools
- content: configure vimgo and goland
- content: configure vscode
- add images for content where applicable
- fix content under languages to be rendered under theleft doc index context
- remove some content that should not be under languages but instead under another context e.g. Security/Fuzz

willpoint committed:
docs: improve content with shared docs and update urls
- update install doc
- add button to detect go version on install doc
- add tutorial for gopls
- implement navigation for prev/next across docs

willpoint committed:
docs/navigation: add context to each nav item
to make navigation clearer

willpoint committed:
remove unneeded println

willpoint committed:
doc/nav: make context of nav item more muted

willpoint committed:
nav/tmpl: make nav func more efficient
- break when item is found

willpoint committed:
homepage: fix download dropdown

odeke-em commented:
- [x] Let's also add visual indicators for previous and next![image](https://github.com/orijtech/website/assets/4898263/cfbd3541-062a-4dad-ad73-17889bc5c806)
- [ ] Our top and bottom jump buttons are needed back ![image](https://github.com/orijtech/website/assets/4898263/f06c1349-7d75-4e17-8ca2-a1acb1977ace)

willpoint committed:
fixes and improvements
- add visual indicators for navigation
- re-order homepage
- fix view to hide 'This page' when page has no outlines
- improve size of burger for mobile menu to make it more accessible

willpoint committed:
mvoe try go on browser to the bottom

willpoint committed:
fix links including external links

willpoint committed:
add more doc content

willpoint committed:
site

willpoint committed:
update documentation to use latest material for 'editors and tools'

willpoint committed:
add omitted vimgo-docs update

willpoint committed:
fix issue with bad links
- only create an ID if one does not exist
while here, use a link without the '.html' suffix to avoid
unnecessary redirects

willpoint committed:
add fixes for missing and broken links
- make links visually accessible by distinguishing them from regular text
  with focus on docs under the Databases section of the left index

willpoint committed:
Fix links and formatting requests per feedback
- FAQ links that reference external pages should open in a new tab
- Code colors should be clearly distinguishable even for inline code blocks
  within sentences
- Fix broken link on the vim-go and Goland docs

willpoint committed:
Add scroll utilities to scroll to top/bottom of the docs page

willpoint committed:
Add a scroll meter to document to tracking reading percentage

odeke-em committed:
all: fix grammatical suggestions

Fixes grammatical suggestions and other latest suggestions.

odeke-em commented on _content/doc/tutorial/getting-started.html:
```diff
@@ -9,7 +9,7 @@
 </p>
 
 <ul>
-  <li>Install Go (if you haven't already).</li>
```
@willpoint kindly help deploy this version as I've made the changes requested. Thank you!

odeke-em committed:
all: add v2 _content concurrently with existing _content

This change creates the ability to have concurrent versions of
the website running so that both websites can run concurrently
and then eventually after full on approvals, we can delete the
incumbent code in _content and rename _content_v2 to _content.

Essentially our redesign now lives in "_content_v2/" and the
current website can continually be developed as is in "_content".

odeke-em commented:
@willpoint I've made a new directory _content_v2 (which we are modifying) and let _content exist as is, this will allow us to make updates as they are to the website then later plumb our routing so that /v2 will point to the content in _content_v2 and when ready, we can then delete _content and replace it with the content in _content_v2. This also allows us to mail CLs that are independent of the website's changes.
